File: 2_direct_line_equation.m
clc; clear; close all;

x1 = 2; y1 = 3;
x2 = 10; y2 = 8;

m = (y2 - y1) / (x2 - x1);
c = y1 - m * x1;

x = x1:x2;
y = m * x + c;

plot(round(x), round(y), 'o');
grid on;
title('Direct Line Equation');

File: 2_direct_line_equation.py
import matplotlib.pyplot as plt

x1, y1 = 2, 3
x2, y2 = 10, 8

m = (y2 - y1) / (x2 - x1)
c = y1 - m * x1

x_points = []
y_points = []

for x in range(x1, x2 + 1):
    y = m * x + c
    x_points.append(x)
    y_points.append(round(y))

plt.plot(x_points, y_points, marker='o')
plt.grid()
plt.title("Direct Line Equation")
plt.show()
